packages/core/src/client/utils.ts

```diff
@@ -5,34 +5,42 @@ import { FetchOptions } from '../models';
 
 export type GraphQLClientOptions = Pick<SitecoreConfigInput, 'api'> & FetchOptions;
 
-/**
- * Creates a new GraphQLRequestClientFactory instance
- * @param {GraphQLClientOptions} options content sdk config
- * @returns GraphQLRequestClientFactory instance
- */
 export const createGraphQLClientFactory = (options: GraphQLClientOptions) => {
-  let clientConfig: GraphQLRequestClientFactoryConfig;
+  let clientConfig: GraphQLRequestClientFactoryConfig | undefined;
 
-  if (options.api?.edge?.contextId) {
+  const { api } = options;
+  const { edge, local } = api ?? {};
+  const isBrowser = typeof window !== 'undefined';
+
+  if (edge?.contextId) {
+    // Real client for server-side rendering / API routes
     clientConfig = {
-      endpoint: getEdgeProxyContentUrl(options.api.edge.contextId, options.api.edge.edgeUrl),
+      endpoint: getEdgeProxyContentUrl(edge.contextId, edge.edgeUrl),
     };
-  } else if (options.api?.local?.apiKey && options.api?.local?.apiHost) {
+  } else if (edge?.clientContextId) {
```

Comment:
```suggestion
  } else if (isBrowser && edge?.clientContextId) {
```
to be extra consistent with the expected behavior